cmd/xl-v1-multipart.go

```diff
@@ -702,22 +702,26 @@ func (xl xlObjects) CompleteMultipartUpload(ctx context.Context, bucket string,
 		// Last part could have been uploaded as 0bytes, do not need
 		// to save it in final `xl.json`.
 		if (i == len(parts)-1) && currentXLMeta.Parts[partIdx].Size == 0 {
-			xlMeta.Parts = xlMeta.Parts[:i] // Skip the part.
+			finalParts = finalParts[:i] // Skip the part.
 			continue
 		}
 
 		// Save for total object size.
 		objectSize += currentXLMeta.Parts[partIdx].Size
 
 		// Add incoming parts.
-		xlMeta.Parts[i] = objectPartInfo{
+		finalParts[i] = objectPartInfo{
 			Number: part.PartNumber,
 			ETag:   part.ETag,
 			Size:   currentXLMeta.Parts[partIdx].Size,
 			Name:   fmt.Sprintf("part.%d", part.PartNumber),
 		}
 	}
 
+	// Save final parts list.
+	xlMeta.Parts = make([]objectPartInfo, len(parts))
```

Comment:
This allocation seems unnecessary - it is alreadying happening in line 713 above.